pkg/pillar/cmd/zedagent/handlecertconfig.go

```diff
@@ -233,14 +237,42 @@ func controllerCertsTask(ctx *zedagentContext, triggerCerts <-chan struct{}) {
 	}
 }
 
+func verifySigningCertNewest(ctx *zedagentContext, certByte []byte) error {
+	block, _ := pem.Decode(certByte)
+	if block == nil {
+		err := fmt.Errorf("verifyCertNewest: pem.Decode() failed")
+		return err
+	}
+	newCert, err := x509.ParseCertificate(block.Bytes)
+	if err != nil {
+		err = fmt.Errorf("verifyCertNewest: x509.ParseCertificate() failed: %w", err)
+		return err
+	}
+	err = zedcloud.LoadSavedServerSigningCert(zedcloudCtx)
+	if err != nil {
+		if errors.Is(err, os.ErrNotExist) {
+			// Certificates were not loaded, accept
+			return nil
+		}
+		err = fmt.Errorf("verifyCertNewest: LoadSavedServerSigningCert() failed: %w", err)
+		return err
+	}
+	oldCert := zedcloudCtx.ServerSigningCert
+	if newCert.NotBefore.Before(oldCert.NotBefore) {
+		err = fmt.Errorf("verifyCertNewest: received old signing certificate, new.NotBefore '%v' < '%v' old.NotBefore",
+			newCert.NotBefore, oldCert.NotBefore)
+		return err
+	}
+
+	return nil
+}
+
 // Fetch and verify the controller certificates. Returns true if certificates have
 // not changed or the update was successfully applied.
 // False is returned if the function failed to fetch/verify/unmarshal certs.
-func requestCertsByURL(ctx *zedagentContext, url string, desc string) bool {
+func requestCertsByURL(ctx *zedagentContext, certURL string, desc string,
+	isSigningCertNewest bool) bool {
```

Comment:
isX sounds like a statement and not a request. checkX (plus a comment above the function definition) would make it more clear.